operator/cmd/mesh/upgrade.go

```diff
@@ -270,25 +266,16 @@ func waitForConfirmation(skipConfirmation bool, l clog.Logger) {
 	}
 }
 
-// checkSupportedVersions checks if the upgrade cur -> tar is supported by the tool
-func checkSupportedVersions(cur, tar, versionsURI string) error {
-	tarGoVersion, err := goversion.NewVersion(tar)
-	if err != nil {
-		return fmt.Errorf("failed to parse the target version: %v", tar)
-	}
-
-	compatibleMap, err := pkgversion.GetVersionCompatibleMap(versionsURI, tarGoVersion)
-	if err != nil {
-		return err
-	}
+var SupportedIstioVersions, _ = goversion.NewConstraint(">=1.6.0, <1.8")
```

Comment:
Should this be `NewConstraint(">=1.4, < 1.8")`?

Reply:
We currently read from versions.yaml. We will match the 1.7.0 block:

```
- operatorVersion: 1.7.0
  operatorVersionRange: ">=1.7.0,<1.8.0"
  supportedIstioVersions: ">=1.6.0, <1.8"
  recommendedIstioVersions: 1.7.0
  k8sClientVersionRange: ">=1.15"
  k8sServerVersionRange: ">=1.15"
```

Which is where this number comes from

Reply:
In Istio 1.4, I believe we agreed to use the Istio Lifecycle of n-1 with the expectation that we would support the release until the next version arrived as per deprecation policy - meaning we needed to look into the future one release to support upgrades.

This contract was [broken in Istio 1.5](https://github.com/istio/operator/pull/710), and it looks like it has been copied to 1.6. Perhaps that is where the root of my misunderstanding is.

Cheers
-steve

Reply:
@howardjohn my apologies on this error. Ever since I broke my glasses (see twitter for the complaining..) and had to revert to an older pair, I am literally seeing double. It makes it a little hard to review these PRs. My eye care professional is closed except for emergencies, and he doesn't consider this an emergency at this time. If I seem to get two lines confused - this is why.

Cheers
-steve